Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,155 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Events;
+using WayOfLife.Model;
+using WayOfLife.View;
+
+namespace WayOfLife
+{
+    public class GameManager : MonoBehaviour
+    {
+        [SerializeField] private ButtonScript restartButton;
+        [SerializeField] private ButtonScript startStopButton;
+        [SerializeField] private ThreeOptionsToggle speedToggle;
+        [SerializeField] private ThreeOptionsToggle sizeToggle;
+        [SerializeField] private ThreeOptionsToggle probToggle;
+        [SerializeField] private InfoText text;
+        [SerializeField] private GameGrid grid;
+        [SerializeField] [Range(0, 500)] private int speedlarge;
+        [SerializeField] [Range(501, 1000)] private int speedmedium;
+        [SerializeField] [Range(1001, 2000)] private int speedsmall;
+        [SerializeField] [Range(0, 50)] private int sizesmall;
+        [SerializeField] [Range(51, 100)] private int sizemedium;
+        [SerializeField] [Range(101, 150)] private int sizelarge;
+        [SerializeField] [Range(0, 33)] private int probsmall;
+        [SerializeField] [Range(34, 66)] private int probmedium;
+        [SerializeField] [Range(67, 100)] private int problarge;
+        private Options options;
+        private bool gameActive;
+        private UnityEvent onRestartClick;
+        private UnityEvent onGameActiveToggle;
+        private UnityEvent onProbChanged;
+        private UnityEvent onSizeChanged;
+        private UnityEvent onSpeedChanged;
+        private int generation = 1;
+        private int aliveCounter;
+
+        void Start()
+        {
+            options = ScriptableObject.CreateInstance<Options>();
```

Comment:
it is recommended to avoid doing new instance of ScriptableObject if you dont need
you can also use [SerializeField] for that